lib/config/decrypt.spec.ts

```diff
@@ -208,5 +216,92 @@ describe('config/decrypt', () => {
         CONFIG_VALIDATION,
       );
     });
+
+    it('handles EC/AES-GCM encryption', async () => {
+      GlobalConfig.set({ privateKey: toBase64(privateEc) });
+      config.encrypted = {
+        token:
+          'eyJrIjp7Imt0eSI6IkVDIiwieCI6IlM0TFhnVUJHMUNlSXVUU1ZiZEg3MHZpSjV6dUxaRFNydFJ2ZUhVTHBBYnBMTXAwQ0tyeVozWm5RN1lKQjh3N2MiLCJ5IjoiZVpQNFZzT1hXZGxRYkNWYWxSakVWNEFzTFBxYmpKdW1mR1dfTzlZbG9HM2dqbzd5enJfUXQtbXl2SG5LdkZheiIsImNydiI6IlAtMzg0In0sImkiOiJNRDVMb2hUMmh2VS9SYXA3IiwibSI6IkptMERpUmNtbEFEU2JjeVdnWmJlUXl4QkdUbU55aVlkV2NDa0tyVnhQZStTdEdMMlBwWT0ifQ==',
+      };
+
+      const res = await decryptConfig(config, 'some/def');
+      expect(res.encrypted).toBeUndefined();
+      expect(res.token).toBe('123');
+      await expect(decryptConfig(config, 'abc/defg')).rejects.toThrow(
+        CONFIG_VALIDATION,
+      );
+    });
+  });
+
+  describe('tryDecryptEc', () => {
+    it('decrypts', async () => {
+      // Generate a new key pair in the browser for each encryption,
+      // this private key will be thrown away after encryption,
+      // so that only Renovate can decrypt with the private key.
+      const browserKeyPair = await crypto.subtle.generateKey(
+        { name: 'ECDH', namedCurve: 'P-384' },
+        false,
+        ['deriveKey'],
+      );
+
+      // `d` is the private key, `x` and `y` are the public key
+      const { d, ...renovatePublicKey } = { ...renovatePrivateKey };
+
+      // import Renovate's public key
+      const importedRenovatePublicKey = await crypto.subtle.importKey(
+        'jwk',
+        renovatePublicKey,
+        { name: 'ECDH', namedCurve: 'P-384' },
+        false,
+        [],
+      );
+
+      // derive shared secret from our private key and Renovate's public key
+      const pw = await crypto.subtle.deriveKey(
+        { name: 'ECDH', public: importedRenovatePublicKey },
+        browserKeyPair.privateKey,
+        { name: 'AES-GCM', length: 256 },
+        false,
+        ['encrypt'],
+      );
+
+      // always use a new IV, should always be 12 bytes
+      const iv = crypto.getRandomValues(new Uint8Array(12));
+
+      // encrypt the message
+      const message = await crypto.subtle.encrypt(
+        { name: 'AES-GCM', iv },
+        pw,
+        // Use TextEncoder instead of Buffer in browser
+        Buffer.from('{"o":"some","v":"123"}'),
+        // new TextEncoder().encode('{"o":"some","v":"123"}'),
+      );
+
+      // prepare the encrypted object
+      const encrypted = {
+        // send our public key
+        k: await crypto.subtle.exportKey('jwk', browserKeyPair.publicKey),
+        // send the IV
+        i: Buffer.from(iv).toString('base64'),
+        // send the encrypted message
+        m: Buffer.from(message).toString('base64'),
+      };
+
+      // encode the encrypted object for easier transport
+      const encCfg = toBase64(JSON.stringify(encrypted));
+
+      // print to update test case above
+      // console.error('encCfg', encCfg);
```

Comment:
I think this is in-between-code to test your PR locally? Should you remove the comment later, when the PR is done?

Reply:
yes, will be renoved on final version